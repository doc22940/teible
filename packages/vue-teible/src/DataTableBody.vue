<template>
  <tbody>
    <tr
      v-for="(d, index) in items" :key="index" :class="{
        [theme.datatable__row]: true,
        [theme['datatable__row--odd']]: index % 2 === 1
      }"
    >
      <data-table-cell
        v-for="(column, columnIndex) in columns" :key="columnIndex" v-bind="column.attrs"
        :item="d" :column="column" :class="{
          [theme.datatable__cell]: true,
          [theme['datatable__cell--last-column']]: columnIndex === columns.length - 1,
          [theme['datatable__cell--last-row']]: index === items.length - 1
        }"
      />
    </tr>
  </tbody>
</template>
<script>
import DataTableCell from './DataTableCell'

export default {
  name: 'DataTableBody',
  components: { DataTableCell },
  props: {
    items: {
      type: Array,
      required: true
    },
    columns: {
      type: Array,
      required: true
    }
  },
  inject: ['$theme'],
  computed: {
    theme () {
      return this.$theme()
    }
  }
}
</script>
