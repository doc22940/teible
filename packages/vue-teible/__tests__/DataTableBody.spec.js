import { shallowMount } from '@vue/test-utils'
import idObj from 'identity-obj-proxy'
import DataTableBody from '../src/DataTableBody.vue'

describe('DataTableBody', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(DataTableBody, {
      propsData: {
        items: [{
          one: 'foo',
          two: 'bar'
        }, {
          one: 'baz',
          two: 'qux'
        }],
        columns: [{
          field: 'one',
          label: 'One'
        }, {
          field: 'two',
          label: 'Two'
        }],
        click: () => {}
      },
      provide: () => ({
        $theme: () => idObj
      })
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
