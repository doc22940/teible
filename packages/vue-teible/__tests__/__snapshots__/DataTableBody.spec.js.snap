// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableBody renders correctly 1`] = `
<tbody>
  <tr class="datatable__row">
    <data-table-cell-stub item="[object Object]" column="[object Object]" class="datatable__cell,[object Object]"></data-table-cell-stub>
    <data-table-cell-stub item="[object Object]" column="[object Object]" class="datatable__cell,[object Object]"></data-table-cell-stub>
  </tr>
  <tr class="datatable__row datatable__row--odd datatable__row--last">
    <data-table-cell-stub item="[object Object]" column="[object Object]" class="datatable__cell,[object Object]"></data-table-cell-stub>
    <data-table-cell-stub item="[object Object]" column="[object Object]" class="datatable__cell,[object Object]"></data-table-cell-stub>
  </tr>
</tbody>
`;
